addons/edi/models/edi_synchronizer.py

```diff
@@ -367,7 +377,13 @@ def execute(self):
                     )
                 )
                 with self.statistics() as stats:
-                    vals_list = [rec.target_values(rec._record_values()) for rec in batch]
+                    vals_list = [
+                        {  k: v
+                            for k,v in rec.target_values(rec._record_values()).items()
+                            if k in safe_eval(rec.changed_fields)
```

Comment:
Good idea using `in`.
We should split changed_fields by `,` in case we have similar two fields in the same record where one is substring of the other one.

Reply:
changed fields is stored as string representation of list of keys, e.g. `['name', 'u_length', 'tracking', 'uom_id', 'uom_po_id', 'u_lot_type']`. Thus `safe_eval` will give us the list back. Hope that should be sufficient?

Reply:
That will be sufficient, good point.